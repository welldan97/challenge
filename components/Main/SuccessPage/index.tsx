import { memo } from 'react';
import styled from 'styled-components';
import { Transaction } from '../../../lib/Transaction';
import Image from '../../lib/Image';
import { Box, Header } from '../styles';

// SECTION: Styles

const Success = styled(Image)`
  margin: 27px auto 11px;
  width: 100px;
`;

const Wrapper = styled.div`
  padding-left: 24px;
  padding-right: 24px;
`;

const Label = styled.div`
  font-size: 16px;
`;

const Amount = styled.div`
  font-size: 28px;
  transform: translateY(-2px);
`;

const Hr = styled.hr`
  border-top: 1px solid #202532;
  opacity: 1;
  margin-top: 12px;
  margin-bottom: 18px;
  width: 142px;
`;

const Address = styled.div`
  transform: translateY(1px);
  margin-bottom: 14px;
`;
// !SECTION

// SECTION: Main

interface Props {
  transaction: Transaction;
}

const SuccessPage = ({ transaction }: Props) => (
  <Box isShrunk>
    <Header>Transaction complete</Header>
    <Wrapper>
      <Success src="/Success.svg" alt="success" width="100" height="100" />
      <Label> You sent</Label>
      <Amount>{transaction.amount} ETH</Amount>
      <Hr />
      <Label>From</Label>
      <Address>{transaction.from}</Address>
      <Label>To</Label>
      <Address>{transaction.to}</Address>
    </Wrapper>
  </Box>
);

// /SECTION
// SECTION: Container

export default memo(SuccessPage);
