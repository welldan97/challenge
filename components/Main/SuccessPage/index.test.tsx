import { render } from '@testing-library/react';
import SuccessPage from '.';

// SECTION: Main

describe('SuccessPage', () => {
  test('render', () => {
    const { container } = render(
      <SuccessPage
        transaction={{ from: 'TEST_FROM', to: 'TEST_TO', amount: '77777' }}
      />,
    );
    expect(container).toMatchSnapshot();
  });
});
