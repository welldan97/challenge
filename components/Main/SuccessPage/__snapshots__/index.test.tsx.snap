// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SuccessPage render 1`] = `
<div>
  <div
    class="sc-da1060d0-0 dVsQgh"
  >
    <header
      class="sc-da1060d0-1 eHCwEr"
    >
      Transaction complete
    </header>
    <div
      class="sc-70be64b8-1 eYoFSo"
    >
      <div
        class="sc-70be64b8-0 jflKBd"
      >
        <span
          style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
        >
          <span
            style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
          >
            <img
              alt=""
              aria-hidden="true"
              src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27100%27%20height=%27100%27/%3e"
              style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
            />
          </span>
          <img
            alt="success"
            data-nimg="intrinsic"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
          <noscript />
        </span>
      </div>
      <div
        class="sc-70be64b8-2 hIOwaP"
      >
         You sent
      </div>
      <div
        class="sc-70be64b8-3 hkrVfx"
      >
        77777
         ETH
      </div>
      <hr
        class="sc-70be64b8-4 ifepvz"
      />
      <div
        class="sc-70be64b8-2 hIOwaP"
      >
        From
      </div>
      <div
        class="sc-70be64b8-5 jEjKnN"
      >
        TEST_FROM
      </div>
      <div
        class="sc-70be64b8-2 hIOwaP"
      >
        To
      </div>
      <div
        class="sc-70be64b8-5 jEjKnN"
      >
        TEST_TO
      </div>
    </div>
  </div>
</div>
`;
